website/docs/concepts/immutability/why_immutability_example.dart

```diff
@@ -0,0 +1,69 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'why_immutability_example.g.dart';
+
+class Fruit {
+  Fruit({this.typeOfFruit = 'Apple'});
+
+  String typeOfFruit;
+}
+
+@riverpod
+Fruit getFruit(GetFruitRef ref) {
```

Comment:
This should respect the codegen/hooks toggles, to be allow using the "manual" provider syntax

Reply:
I believe I've addressed this now. Please take a look and let me know what you think. The relevant files can be found under `website/docs/concepts/immutability`.